visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts

```diff
@@ -112,22 +110,15 @@ export class CodeMapMouseEventService
 			this.threeCameraService.camera.updateMatrixWorld(false)
 
 			if (this.threeSceneService.getMapMesh()) {
-				this.intersectionResult = this.threeSceneService
+				let intersectedBuilding: CodeMapBuilding = this.threeSceneService
 					.getMapMesh()
 					.checkMouseRayMeshIntersection(this.mouse, this.threeCameraService.camera)
-
 				const from = this.threeSceneService.getHighlightedBuilding()
-				let to = null
-
-				if (this.intersectionResult.intersectionFound) {
-					to = this.intersectionResult.building
-				} else {
-					to = this.highlightedInTreeView
-				}
+				const to = intersectedBuilding ? intersectedBuilding : this.highlightedInTreeView
 
 				if (from !== to) {
 					this.unhoverBuilding()
-					if (to) {
+					if (to !== null) {
```

Comment:
`intersectedBuilding` can be undefined. This new if-statement would always be true, because `undefined !== null` and `A_BUILDING !== null` is both true, isn't it?

Reply:
True, this was not a useful change. I undid it. But still, highlighting worked as expected also with the distribution bar.